src/test/java/org/pesmypetcare/webservice/service/MealServiceTest.java

```diff
@@ -0,0 +1,150 @@
+    private static MealEntity mealEntity;
+    private static String owner;
+    private static String petName;
+    private static String date;
+    private static String date2;
+    private static String field;
+    private static String value;
+
+    @Mock
+    private MealDao mealDao;
+
+    @InjectMocks
+    private MealService service = new MealServiceImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        mealList = new ArrayList<>();
+        mealEntity = new MealEntity();
+        date = "2020-02-13T10:30:00";
+        date2 = "2021-02-13T10:30:00";
+        owner = "Pepe05";
+        petName = "Camper";
+        field = "mealName";
+        value = "Asparagus";
+    }
+
+    @Test
+    public void shouldReturnNothingWhenMealCreated() {
+        service.createMeal(owner, petName, date, mealEntity);
+        verify(mealDao).createMeal(isA(String.class), isA(String.class), isA(String.class), isA(MealEntity.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenMealDeleted() {
+        service.deleteByDate(owner, petName, date);
+        verify(mealDao).deleteByDate(isA(String.class), isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenAllMealsDeleted() throws DatabaseAccessException {
+        service.deleteAllMeals(owner, petName);
+        verify(mealDao).deleteAllMeals(isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenAllMealDeleteFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(mealDao).deleteAllMeals(any(String.class), isA(String.class));
+            service.deleteAllMeals(owner, petName);
+        }, "Should return a database access exception when a meal deletion fails");
+    }
+
+    @Test
+    public void shouldReturnMealEntityWhenMealRetrieved() throws DatabaseAccessException {
+        when(mealDao.getMealData(owner, petName, date)).thenReturn(mealEntity);
+        MealEntity meal = service.getMealData(owner, petName, date);
+        assertSame(mealEntity, meal, "Should return a meal entity");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetMealRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(mealDao).getMealData(any(String.class), any(String.class),
+                isA(String.class));
+            service.getMealData(owner, petName, date);
+        }, "Should return an exception when retrieving a meal fails");
+    }
+
+    @Test
+    public void shouldReturnMealEntityListWhenGetSetOfMealsRetrieved() throws DatabaseAccessException {
+        when(mealDao.getAllMealData(owner, petName)).thenReturn(mealList);
+        List<Map<String, Object>> list = service.getAllMealData(owner, petName);
+        assertSame(mealList, list, "Should return a list of meal entities");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "Should return a list of meal entities" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5301962)